naucse/models.py

```diff
@@ -271,6 +271,27 @@ def materials(self):
         return materials
 
 
+    def get_coverpage_content(self, run, coverpage, app):
+        session_template_loader = jinja2.FileSystemLoader(path.join(app.root_path, '..', 'runs'))
+        env = app.jinja_env.overlay(loader=session_template_loader)
+        name = '{}/sessions/{}/{}.md'.format(run.slug, self.slug, coverpage)
+
+        try:
+            template = env.get_template(name)
+        except TemplateNotFound:
+            abort(404)
```

Comment:
You need to import `abort` to use it.
But please don't: `abort` is a Flask function; it doesn't belong in the model. Throw an exception (`LookupError` for example), and catch it in routes to abort.